modules/media/php/mediafileprovisioner.class.inc

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * This file implements a data provisioner to get all possible files
+ * for the media menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\media;
+
+/**
+ * This class implements a data provisioner to get all possible files
+ * for the media menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class MediaFileProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
+{
+    /**
+     * Create a MediaFileProvisioner, which gets files for the meida
+     * menu table.
+     */
+    function __construct()
+    {
+        parent::__construct(
+            "SELECT m.file_name,                                                   
+                    (SELECT PSCID from candidate WHERE CandID=s.CandID) as pscid,
+                    s.Visit_label as visit_label,
+                    l.language_label,
+               			(SELECT Full_name FROM test_names WHERE Test_name=m.instrument),
+               			s.CenterID as site,
+               			m.uploaded_by,
+               			m.date_taken,
+               			substring(m.comments, 1, 50) as comments,
+               			m.date_uploaded,
+               			m.file_type,
+               			s.CandID as cand_id,
+               			s.ID as session_id,
+               			m.hide_file,
+               			m.id
```

Comment:
What are you trying to do with this convoluted/slow sub-subselect query that you couldn't just do with a join?

Reply:
This is the same as the previous query, I think the columns are just misaligned so it looks like several sub-queries. I'll try and fix the formatting!